src/trace.c

```diff
@@ -1051,14 +1088,6 @@ trace_add_path (const void *parent,
 		nih_hash_add (path_hash, &entry->entry);
```

Comment:
In the new logic, we're adding only either of the original path or the resolved path. How about adding both? I think we can factor out the part of hash entry search and addition to a helper function, then move the new realpath logic to where we originally had `lstat` check, inside which we can call the hash helper for the resolved path then.

Reply:
Added a helper function to handle duplicate check.